File: 674. Longest Continuous Increasing Subsequence/solution.py
class Solution(object):
  def findLengthOfLCIS(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    maxSub = 0
    counter = 1
    for index in range(1, len(nums)):
      prevNum = nums[index - 1]
      num = nums[index]
      if prevNum >= num:
        maxSub = max(maxSub, counter)
        counter = 1
      else:
        counter += 1
    return max(maxSub, counter)

File: 674. Longest Continuous Increasing Subsequence/solution.cpp
class Solution {
public:
  int findLengthOfLCIS(vector<int>& nums) {
    int maxSub = 0;
    int counter = 1;
    for (int index = 1; index < nums.size(); index++){
      int prevNum = nums[index - 1];
      int num = nums[index];
      if (prevNum >= num){
        maxSub = max(maxSub, counter);
        counter = 1;
      }
      else{
        counter += 1;
      }
    }

    return max(maxSub, counter);
  }
};